### PR TITLE
Update CRS and measurements settings in Project properties dialog

### LINKED ISSUE #1624: [needs-docs] Rename 'Parameters:' ellipsoid option to 'Custom'
Original commit: https://github.com/qgis/QGIS/commit/797abbd718885d56e3e3a8390cf032dde4f827be by nyalldawson

To better respect UI standards

### LINKED ISSUE #1724: [FEATURE] Allow setting project CRS to "no projection"
Original commit: https://github.com/qgis/QGIS/commit/899888d1c533df37b8c60a1cc9532d2055710652 by nyalldawson

This replaces the older "OTF off" setting. If a project has
no projection set, then the following things happen:

- no reprojection is done while rendering - features are just
drawn using their raw coordinates
- ellipsoid, distance and area units, and coordinate display
are locked out.
- all measurements are done in unknown map units, and no conversion
is possible.

As well as addressing the use cases in which users disabled
OTF reprojection, this setting may also be useful for users
who use QGIS to create "non-earth" maps, such as RPG or fantasy
maps, or for microscopic analysis, or some other local reference
system not related to earth based projections.

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #1724, fixes #1624, refs #2167, refs #1623

### CODE REVIEW

File: source/docs/user_manual/introduction/qgis_configuration.rst [hunk 1]
```diff
@@ -814,10 +818,31 @@ this is not enforced. See :ref:`vector layer metadata properties
...
-The :guilabel:`CRS` tab enables you to choose the Coordinate Reference
-System for this project, and to enable on-the-fly re-projection of raster and
-vector layers when displaying layers from a different CRS. For more information
-on projection's handling in QGIS, please read :ref:`label_projections` section.
+The |crs| :guilabel:`CRS` tab helps you set the coordinate reference system
+to use in this project. It can be:
+
+* |checkbox| :guilabel:`No projection (or unknown/non-Earth projection)`:
+  this setting can be used to guess a layer CRS or when using QGIS for
+  non earth uses like rpg maps, building mapping or microscopic stuff.
```
Comment (yjacolin) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: **rpg** means to be explained, isn't it?

File: source/docs/user_manual/introduction/qgis_configuration.rst [hunk 2]
```diff
@@ -814,10 +818,31 @@ this is not enforced. See :ref:`vector layer metadata properties
...
+* or an existing coordinate reference system that can be *geographic*, *projected*
+  or *user-defined*. A preview of the CRS extent on earth is displayed to
+  help you select the appropriate one.
+  Layers added to the project are on-the-fly translated to this CRS in order
+  to overlay them regardless their original CRS. Use of units, ellipsoid setting
+  are available and make sense and you can perform calculations accordingly.
+
+The |crs| :guilabel:`CRS` tab also helps you control the layers reprojection
+settings by configuring the datum transformation preferences to apply in the
+current project. As usual, these override any corresponding global setting.
```
Comment (yjacolin) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: setting**s**?

File: source/docs/user_manual/introduction/qgis_configuration.rst [hunk 3]
```diff
@@ -797,10 +801,31 @@ this is not enforced. See :ref:`vector layer metadata properties
...
+* |checkbox| :guilabel:`No projection (or unknown/non-Earth projection)`:
+  this setting can be used to guess a layer CRS or when using QGIS for non
+  earth uses like role-playing game maps, building mapping or microscopic stuff.
+  In this case:
+
+  * no reprojection is done while rendering the layers: features are just drawn
+    using their raw coordinates;
+  * the ellipsoid is locked out and forced to ``None/Planimetric``;
+  * the distance and area units, and the coordinate display are locked out and
+    forced to unknown "map units"; all measurements are done in unknown map units,
```
Comment (SrNetoChan) [inline source/docs/user_manual/introduction/qgis_configuration.rst]: unknown "map units" --> "unknown units"

**Comment by DelazJ** [comment]
I expected to update the https://docs.qgis.org/testing/en/docs/user_manual/working_with_projections/working_with_projections.html and get both reviewed by @nyalldawson but never got time/energy to dive into this

**Pull request merged**